### DISCUSSION TITLE
Self rotating Spinner?

### DISCUSSION THREAD

**Discussion by yanshay** [slint-ui/slint#6945] [Q&A]
Slint has a `spinner` but it seems like I need to manage its progress.
I can set a timer and drive the spinner but I wonder if there's a simpler way.
In HTML I can just define a spinner as css and it spins. Is there a way to do that with Slint spinner as well?

**Comment by hunger** [comment]
Is this what you are looking for?

```slint
import { Spinner } from "std-widgets.slint";
export component Demo {
    VerticalLayout {
        Spinner {
            indeterminate: true;
        }
    }
}
```

[Obligatory Slintpad link](https://slintpad.com/?snippet=import+%7B+Spinner+%7D+from+%22std-widgets.slint%22%3B%0Aexport+component+Demo+%7B%0A++++VerticalLayout+%7B%0A++++++++Spinner+%7B%0A++++++++++++indeterminate%3A+true%3B%0A++++++++%7D%0A++++%7D%0A%7D%0A&style=native)

**Reply by yanshay** [reply]
Thanks. That's the idea, it looks different from the standard HTML spinner but that's the idea.
However, I checked the source for spinner, and it uses `Path` so won't work on an embedded device, so I can't use it.
That lack of `Path` or svg on embedded is pretty limiting in what one can do with Slint on embedded device.

I also see that it uses `animation-tick()`, which if I remember correctly, in an async environment will cause the event loop to run like a busy loop, so intensive cpu usage taking resources from the rest of the system.
Probably better to use a timer instead.

**Reply by hunger** [reply]
`animation-tick()` should work just fine in that use case: It does keep the event loop busy, but only while animations are running:-)

**Reply by yanshay** [reply]
This spinner would run many times when there is processing which may need fast response or cpu and this behavior could slow things down.

In an application I wrote in the past due to event loop taking processing time during animations I had to move wifi processing to the 2nd core one the Esp32 so not to miss udp packets. I don't know why that happened, maybe due to non optimal wifi code, but best to take as little cpu on these systems.
